src/Sylius/Behat/Client/ResponseChecker.php

```diff
@@ -132,18 +132,54 @@ public function hasItemWithValue(Response $response, string $key, $value): bool
         return false;
     }
 
-    /** @param string|int $value */
-    public function hasSubResourceWithValue(Response $response, string $subResource, string $key, $value): bool
-    {
+    public function hasValueInAnySubresourceObjectCollection(
+        Response $response,
+        string $subResource,
+        string $key,
+        int|string $expectedValue,
+    ): bool {
         foreach ($this->getResponseContentValue($response, $subResource) as $resource) {
```

Comment:
```suggestion
        $resources = $this->getResponseContentValue($response, $subResource);
        foreach ($resources as $resource) {
```
To ease debugging a bit